### DISCUSSION TITLE
Lambda裡return之詢問😅

### DISCUSSION THREAD

**Discussion by JunTingLin** [1101-GS4719-C/Python-Community-GS4719-C#578] [Q&A]
# Lambda裡return之詢問😅

## 源由
上課練習Quiz15，發現"有關Python匿名函數，不可使用return敘述"，這句是錯的，
代表Lambda還是可以下return敘述。
所以我就來試試🔽

## 正常標準Lambda

```
r = operate_on(10,40,lambda a,b:a-b)   
print('operate_on(10,40,Lambda_function) =', r)
```
## 加上return回傳a-b
下面這樣就...爆了，回我語法錯誤😆
```
r = operate_on(10,40,lambda a,b: return a-b)   
print('operate_on(10,40,Lambda_function) =', r)
```

有大大知道為甚麼嗎?

那return搭配Lambda要如何使用呢?

**Comment by chia-kai-chang** [comment]
抱歉那題題目有瑕疵，Lambda函數確實**不能使用**`retuen`，只能有**表達式**，不能有`return`
以Python官方Doc為準：https://docs.python.org/3/reference/expressions.html#lambda
